### ISSUE TITLE
Reproducing the alpamayo paper results for reasoning metrics

### ISSUE THREAD

**Issue by rdesc** [NVlabs/alpamayo1.5#29]
We have a question regarding our seemingly poor performance on reasoning metrics with the alpamayo models. We evaluated `Alpamayo-R1-10B` and `Alpamayo-1.5-10B` (no extra fine-tuning) on the OOD reasoning subset of PhysicalAI-AV. Trajectory metrics line up with Tables 6–7 from the paper, whereas reasoning scores are quite bad. Let me first be explicit about how we compute everything.

### How each number is computed

**Sampling / "most-likely" rollout.** Per event we draw K=6 joint (CoC + trajectory) samples at temperature 0.6, top_p 0.98 (repo defaults), unconditioned (no route/nav). The **medoid** — the sample minimizing summed L2 trajectory distance to the other 5 — is taken as the "most-likely" rollout, and both reasoning grading and consistency are computed on that single medoid CoC / medoid trajectory.

**Reasoning [0,1].** `wayveai/Lingo-Judge` text grader, medoid pred CoC vs ground truth CoC (text-only).

**Reasoning [0–5].** Our reconstruction of the §5.3.2 LLM-judge rubric, run on a vision judge (`nvidia/Cosmos-Reason2-8B`) that sees the t0 multi-cam image + GT CoC + predicted CoC.

**CoC-Action consistency (binary {0,1}).** Our reconstruction of §5.3.2. We classify the medoid trajectory into a meta-action over a 3 s decision window across **two axes** — longitudinal from Δv (speed-up / slow-down / steady / reverse) and lateral from net heading change + peak lateral offset (left / right / straight) — and parse the medoid CoC text into an intended per-axis decision. `R=1` iff both axes the reasoning explicitly states matches the trajectory; unparseable reasoning → `R=0`. Detectors are **ego-xy-only (no lane/map)**, and matching is at coarse axis granularity since the CoC text rarely states maneuver magnitude. Example below which visualizes our reimplementation of this metric: 
<img width="1532" height="536" alt="Image" src="https://github.com/user-attachments/assets/477596c0-1ad3-4039-b011-8bd143de4469" />

**Consistency ceiling (GT CoC + GT trajectory).** Running the *same* consistency pipeline on the **GT CoC text** and the **ego GT future trajectory** gives **0.825** on val. It falls short of 1.0 because PAI-AV has **no map / lane geometry** (only egomotion + obstacle labels + camera/lidar/radar), so our meta-action detectors are ego-motion-only and cannot measure a maneuver relative to a lane. Our predicted consistency should be read **relative to it, not to 1.0**. Example below which visualizes a failure case of our reimplementation of CoC-Action consistency. 

<img width="1570" height="536" alt="Image" src="https://github.com/user-attachments/assets/1f98a373-88c0-4b8b-95ad-53247b685f1e" />

### Results 

| Split | Model | Reasoning [0,1] ↑ | Reasoning [0–5] ↑ | CoC-Action Consist. ↑ | Consist. ceiling (GT CoC + GT traj) |
|---|---|---|---|---|---|
| **val** (n=349) | Alpamayo 1.5 | 0.706 | 3.17 | 0.344 | 0.825 |
| **val** (n=349) | Alpamayo R1 | 0.528 | 3.09 | 0.381 | 0.825 |
| **train** (n=1728) | Alpamayo 1.5 | 0.682 | 3.37 | 0.447 | 0.798 |
| **train** (n=1728) | Alpamayo R1  | 0.524 | 3.29 | 0.436 | 0.798 |

### Questions

1. **Is the very similar performance for CoC-action consistency between 1.5 and R1 expected for the released weights?** 
In the table we see that 1.5 ≈ R1 on CoC-action consistency, on both splits, despite 1.5 being RL post-trained and the paper attributing the RL gain to consistency.

2. **Are the reasoning grading scores and the CoC-action consistency scores expected?**
Our [0–5] grading sits ~3.1–3.4 vs the paper's 4.5 (Table 9). Our GT-CoC-vs-GT-trajectory consistency ceiling is ~0.8 and we get scores of ~0.35-0.45. We'd like to know whether the released checkpoints are expected to reproduce the Table 9 numbers at all. Even qualatively, we find that the planned trajectory is frequently inconsistent with the CoC.

3. **For Table 9's reasoning/consistency, how is the "most-likely" rollout selected among samples (we used the medoid of 6) ?**

Thanks!

**Comment by lucamodica** [comment]
Hi, a bit unrelated but can I ask you how did you output the meta-action from Alpamayo1.5? With default parameter values, It does not seem to generate any.

**Comment by rdesc** [comment]
@lucamodica I had to write some logic to extract a meta-action from both the CoC and the trajectory. The provided models indeed do not support meta action generation

**Comment by summer4272** [comment]
Could you use the nav_text ? How to get the information?

**Comment by rdesc** [comment]
@lucamodica heads up https://github.com/NVlabs/alpamayo-coc-autolabeler provides code for extracting meta action from the trajectory, albeit it seems like it would work better if the map lane graph / lane information is provided

**Comment by lucamodica** [comment]
just saw it now thank you a lot!

**Comment by PushpakAg** [comment]
Have you checked if the way you're computing the reasoning metrics aligns with the methods described in the original paper? There might be differences in implementation details that could affect the results.
